### PR TITLE
Refactor habit store to include habit class

### PR DESCRIPTION
**Pull request by devinjameson**
Why: we would like to create a Habit class so we can update habits directly
instead of telling the habitStore to update a particular habit.

```typescript
habitStore.updateHabit(nextHabit)

// vs

habit.update({color})
```

This commit:
Creates a Habit class
Makes general refactors to improve code quality

### CODE REVIEW

File: src/storage.ts [hunk 1]
```diff
@@ -15,10 +15,8 @@ export const storeData = async (
   }
 }
 
-export const storeHabitStore = async (
-  habitStore: Record<string, Habit.Habit>,
-): Promise<void> => {
-  storeData("Habit", JSON.stringify(habitStore))
+export const storeHabits = async (habits: Habit[]): Promise<void> => {
```
Comment (johnschoeman) [inline src/storage.ts:18]: should we be versioning this?

File: src/storage.ts [hunk 2]
```diff
@@ -32,11 +30,11 @@ export const getData = async (
   }
 }
 
-export const getHabitStore = async (): Promise<Record<string, Habit.Habit>> => {
-  const rawHabit = await getData("Habit")
-  if (rawHabit) {
-    return JSON.parse(rawHabit)
+export const getHabits = async (): Promise<HabitParams[]> => {
+  const rawHabits = await getData("Habit")
```
Comment (johnschoeman) [inline src/storage.ts:34]: Do we need to handle both versions here for backwards compatibility?

**Review by johnschoeman — COMMENTED** [review]
Updating the habits directly definitely seems like a better fit for the mobx style!

File: src/storage.ts (hunk 2 above)
Comment (devinjameson) [reply src/storage.ts:34]: Totally, would be a good thing to pair on.

**Pull request merged**